pkg/lib/common.ts

```diff
@@ -26,32 +28,32 @@ export function in_array(array, val) {
     return false;
 }
 
-export function is_function(x) {
+export function is_function(x: unknown): x is (...args: unknown[]) => unknown {
```

Comment:
What's that weird `x is (...args: unknown[])` thing? and why is the return value `unknown`? It should be `boolean`, and the function can never return anything else.

Reply:
The return value isn't unknown.  This is a user-defined type guard.  The return value is `x is (...args: unknown[]) => unknown`, which means "the return type is a function with unknown arguments and unknown return value", which accurately describes what this function does.

Reply:
Ah, the `=> unknown` isn't the return value of `is_function`, that whole contraption is. But that's not the case -- the return value is a `boolean`, not a function.

Reply:
`: x is [type]` is a special case of `: boolean`.  See https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

We could write this as `: boolean` but then we wouldn't benefit from type narrowing when we use it.

The idea is when you write

```ts
function call_if_function(something: unknown) {
    something();
}
```

That's an error.  But if you first

```ts
if (is_function(something)) {
    ...
}
```

then at least inside of the block you know that it's a function.  You still can't really call it, since you don't know the argument type, of course...

So maybe it makes sense to just use `boolean` here after all?  I sort of like the documentation-via-return-type...

Reply:
Ah, on the fourth reading I finally understand that type signature. This is more of a "TS newbie" issue then. Now it makes sense, and getting at least a partial answer in the type is nice. Please feel free to just resolve this and leave it as it is.